src/hooks/useResponsiveLayout.ts

```diff
@@ -12,10 +13,20 @@ type ResponsiveLayoutResult = {
  */
 export default function useResponsiveLayout(): ResponsiveLayoutResult {
     const {isSmallScreenWidth} = useWindowDimensions();
-    const state = navigationRef?.current?.getRootState();
-    const lastRoute = state?.routes?.at(-1);
-    const lastRouteName = lastRoute?.name;
-    const isInModal = lastRouteName === NAVIGATORS.LEFT_MODAL_NAVIGATOR || lastRouteName === NAVIGATORS.RIGHT_MODAL_NAVIGATOR;
+    const isDisplayedInModal = useCallback(() => {
+        const state = navigationRef?.current?.getRootState();
+        const lastRoute = state?.routes?.at(-1);
+        const lastRouteName = lastRoute?.name;
+        return lastRouteName === NAVIGATORS.LEFT_MODAL_NAVIGATOR || lastRouteName === NAVIGATORS.RIGHT_MODAL_NAVIGATOR;
+    }, []);
```

Comment:
Actually, this function can be statically defined outside of component lifecycle as requires no dependency.